File: backend/PlugPlay/PlugPlay.Domain/Entities/Order.cs
using PlugPlay.Domain.Enums;

namespace PlugPlay.Domain.Entities;

public class Order
{
    public int Id { get; set; }

    public int? UserId { get; set; }

    public DateTime OrderDate { get; set; }

    public OrderStatus Status { get; set; }

    public decimal TotalAmount { get; set; }

    public decimal? DiscountAmount { get; set; }

    public DeliveryMethod DeliveryMethod { get; set; }

    public PaymentMethod PaymentMethod { get; set; }

    public int? DeliveryAddressId { get; set; }

    public PaymentStatus PaymentStatus { get; set; }

    public long TransactionId { get; set; }

    public DateTime? PaymentCreated { get; set; }

    public DateTime? PaymentProcessed { get; set; }

    public string PaymentFailureReason { get; set; }

    public DateTime UpdatedAt { get; set; }

    public User User { get; set; }

    public UserAddress DeliveryAddress { get; set; }

    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}

File: frontend/src/models/Order.ts
import {Address} from "./Address.ts";
import OrderItem from "./OrderItem.ts";
import DeliveryMethod from "./enums/DeliveryMethod.ts";
import OrderStatus from "./enums/OrderStatus.ts";
import PaymentMethod from "./enums/PaymentMethod.ts";
import PaymentStatus from "./enums/PaymentStatus.ts";

// Extended OrderItem with UI-specific properties
export interface OrderItemWithDetails extends OrderItem {
    productName?: string;
    price?: number;
}

export interface Order {
    id: number;
    userId: number;
    orderDate: string;
    status: OrderStatus;
    totalAmount: number;
    discountAmount: number;
    deliveryMethod: DeliveryMethod;
    paymentMethod: PaymentMethod;
    deliveryAddressId: number;
    paymentStatus: PaymentStatus;
    transactionId: number;
    paymentCreated: string;
    paymentProcessed: string;
    paymentFailureReason: string | null;
    updatedAt: string;
    deliveryAddress: Address;
    orderItems: OrderItemWithDetails[];
}